src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -60,16 +62,29 @@ public DiscountResponseVO create(DiscountRequestVO request) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public DiscountResponseVO update(Long id, DiscountRequestVO request) {
-        Discount discount = discountRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", id));
+        Discount discount = getDiscountById(id);
+        String imageUrl = discount.getImageUrl();
+
         discountMapper.updateEntity(request, discount);
         discountRepository.save(discount);
+
+        if (!imageUrl.equals(request.getImageUrl())) {
```

Comment:
imageUrl can be null

Reply:
I added condition. The vendor has a non-null constraint, but Discount doesn't have